java/memory/src/main/java/org/apache/arrow/memory/AllocationManager.java

```diff
@@ -172,10 +154,10 @@ void release(final BufferLedger ledger) {
         // the only <allocator, reference manager> mapping was for the owner
         // which now has been removed, it implies we can safely destroy the
         // underlying memory chunk as it is no longer being referenced
-        ((BaseAllocator)oldLedger.getAllocator()).releaseBytes(size);
+        ((BaseAllocator)oldLedger.getAllocator()).releaseBytes(getSize());
```

Comment:
why the change?

Reply:
This is to keep up with the current behaviour. Following is the constructor code of current `AllocationManager`:
https://github.com/apache/arrow/blob/21a4ffe7411ac62c97d636a478481d96c26380a3/java/memory/src/main/java/org/apache/arrow/memory/AllocationManager.java#L73-L84

This allows an AllocationManager allocates different bytes of memory comparing to desired. So at here, `size` is desired size passed by constructor, while `getSize()` should be the actual size. I'll do some name changes in case of confusing.

Reply:
If size and getSize() are different, maybe this should be made explicit in the JavaDoc?

Reply:
yeah, maybe call one requested_size and one size (or similar)